libraries/core_libs/storage/src/storage.cpp

```diff
@@ -417,14 +417,18 @@ std::optional<SortitionParamsChange> DbStorage::getParamsChangeForPeriod(uint64_
   return SortitionParamsChange::from_rlp(dev::RLP(it->value().ToString()));
 }
 
-void DbStorage::clearPeriodDataHistory(uint64_t period, uint64_t dag_expiry_period, bool force) {
+void DbStorage::clearPeriodDataHistory(uint64_t period, uint64_t dag_expiry_level, bool force) {
   // Actual history size will be between 100% and 110% of light_node_history_ to avoid deleting on every period
   if (is_light_node_ && ((period % (std::max(light_node_history_ / 10, (uint64_t)1)) == 0) || force) &&
-      period > light_node_history_) {
+      period > light_node_history_ && dag_expiry_level > 0) {
```

Comment:
dag_expiry_level > 0? I think that will delete DAG blocks that have not expired in db_->DeleteRange(write_options_, handle(Columns::period_data), toSlice(start), toSlice(end));

Reply:
For example, dag_expiry_level = 1, we should only remove DAG blocks in level 1.
But proposal_period will be 0, in proposal_period 0 DAG levels should be [0, 100].
In the code, actually removed DAG levels to 100, I think that's wrong

Reply:
This line prevents deleting non-expired data:

// This prevents deleting any data needed for dag blocks proposal period, we only delete periods for the expired dag
    // blocks
    const uint64_t end = std::min(period - light_node_history_, *proposal_period);

Reply:
I have made modifications and now I delete based on this period:
const auto proposal_period = db_->getProposalPeriodForDagLevel(dag_expiry_level - max_levels_per_period_ - 1);
So I am getting the proposal period of dag_expiry_level - max_levels_per_period_ - 1 which ensures that this period cannot have a dag block of level dag_expiry_level  or higher